src/analysis/vector/geometry_checker/qgsfeaturepool.cpp

```diff
@@ -33,6 +33,7 @@ QgsFeaturePool::QgsFeaturePool( QgsVectorLayer *layer )
   , mGeometryType( layer->geometryType() )
   , mFeatureSource( std::make_unique<QgsVectorLayerFeatureSource>( layer ) )
   , mLayerName( layer->name() )
+  , mIsMemoryLayer( layer->dataProvider()->name() == QStringLiteral( "memory" ) )
```

Comment:
Slightly safer, won't crash if the layer is invalid:

```suggestion
  , mIsMemoryLayer( layer->providerType() == QStringLiteral( "memory" ) )
```